### PR TITLE
feat: add bulk action confirmation dialogs and keep checkboxes always selectable

### LINKED ISSUE #83: Bulk Accept/Reject missing confirmation dialog and checkboxes disabled
**Description**

The Bulk Accept and Bulk Reject buttons on the Applications page submit immediately without a confirmation prompt. Since these actions send emails to applicants, an accidental click can mass-notify people. Additionally, once an application is accepted or rejected (individually or via bulk), its checkbox becomes greyed out and unselectable, preventing organizers from revisiting that decision later using bulk actions.

**Current Behavior**

1. Clicking "Bulk Accept" or "Bulk Reject" immediately submits the form with no confirmation dialog.
2. After an application's status changes from "Pending" to "Accepted" or "Rejected", its row checkbox is disabled and cannot be selected again.
3. The backend only processes applications that are still in "Pending" status during bulk actions.

Image

**Steps to Reproduce**

1. Go to TeamShifts → People → Applicants.
2. Select a few pending applications and click "Bulk Accept" — notice it fires instantly with no confirmation.
3. Observe the accepted applications now have greyed-out checkboxes.
4. Try to select those same applications to bulk-reject them — the checkboxes are disabled and cannot be clicked.

**Expected Behavior**

1. Clicking "Bulk Accept" or "Bulk Reject" should show a confirmation dialog warning that emails may be sent to applicants, requiring explicit confirmation before proceeding.
2. Checkboxes should remain selectable at all times regardless of the application's current status, allowing organizers to change decisions later.
3. The backend should allow bulk status changes on applications in any status (not just pending), skipping only those already in the target status to avoid duplicate emails.

**Notes**

- Individual per-row status dropdowns already have confirmation dialogs (added in #74); this brings parity to the bulk action buttons.
- The fix should also add an alert when no applications are selected and the user clicks a bulk button.

### PR DESCRIPTION
**Pull request by MukundC25**
#### Summary

Adds confirmation dialogs to Bulk Accept/Reject buttons on the applications page and removes the disabled state from checkboxes so organizers can revisit decisions at any time.

#### Problem

Previously, once an application was accepted or rejected, its checkbox was greyed out and could not be included in bulk actions again. Additionally, bulk actions had no confirmation step, risking accidental mass status changes and email sends.

#### Changes

- **`applications.js`** — Added `setupBulkActions()` with `showConfirmDialog` before form submission; removed `:not([disabled])` filter from select-all logic.
- **`applications.html`** — Buttons changed to `type="button"` with JS-driven submission; removed `disabled` condition on checkboxes; added `aria-label` for accessibility; added hidden `action` input.
- **`views.py`** — `BulkApplicationStatusView` now accepts applications in any status, excluding only those already in the target status to avoid duplicate lifecycle emails.
- **`tests/test_bulk_application_status.py`** — 6 new tests covering bulk accept, reject, re-decision, same-status skip, empty selection, and invalid action.

https://github.com/user-attachments/assets/0118691d-0265-4d2a-bda0-156f3a59c184

#### Risk and Rollback

Low risk. The change is additive (confirmation UX) and loosens a restriction (checkbox availability). Rollback is a simple revert.

Closes #83 

## Summary by Sourcery

Add confirmation-driven bulk application status changes that work on previously decided applications while avoiding duplicate updates.

New Features:
- Add confirmation dialogs for bulk accept and reject actions on the applications page.
- Allow bulk actions to be applied to applications regardless of their current status, while skipping those already in the target status.

Enhancements:
- Keep application checkboxes always selectable and include them in select-all, improving flexibility for organizers.
- Adjust bulk action buttons and form wiring to use a hidden action field and better align with the new confirmation flow.
- Improve accessibility of application selection checkboxes with aria-labels.

Tests:
- Add integration-style tests covering bulk accept, reject, re-deciding applications, skipping already-updated applications, empty selections, and invalid actions.

### CODE REVIEW

File: teamshifts/views.py [hunk 1]
```diff
@@ -636,24 +636,24 @@ def post(self, request, *args, **kwargs):
                 messages.warning(request, _("No applications selected."))
```
Comment (Rachit7168) [inline teamshifts/views.py:636]: there is a classic Django pitfall here. When you use .update() on a queryset, it directly updates the database, but it does not update the model instances that you already loaded into memory.

Because you evaluated the queryset into a list (apps = list(...)) before the .update() call, the app objects in your for app in apps: loop will still hold their old status (e.g., PENDING). If your email queueing logic relies on app.status to determine which email template to send, it will read this stale state and could send the wrong email.

Recommendation: Explicitly update the in-memory state of the objects inside your loop before triggering the emails

File: teamshifts/static/teamshifts/applications.js [hunk 2]
```diff
@@ -41,13 +41,43 @@ function setupSelectAll() {
     }
 
     selectAll.addEventListener("change", () => {
-        document.querySelectorAll(".app-checkbox:not([disabled])").forEach((checkbox) => {
+        document.querySelectorAll(".app-checkbox").forEach((checkbox) => {
```
Comment (Rachit7168) [inline teamshifts/static/teamshifts/applications.js:44]: Please double-check that window.showConfirmDialog is actually defined globally in your frontend environment. In eventyay/pretix core, confirmation modals are typically invoked using bootbox.confirm(...) or by simply adding data-confirm="Are you sure?" attributes directly to the HTML buttons (which the core JS automatically intercepts and handles for you).

If showConfirmDialog is a custom Promise wrapper you or your team introduced in a recent PR, you can completely ignore this!

**Review by Rachit7168 — APPROVED** [review]
Thanks for working on this. Please address

**Review by Aqil-Ahmad — COMMENTED** [review]
Are all of these tests necessary? feels a bit heavy for regression

File: teamshifts/views.py (hunk 1 above)
Comment (MukundC25) [reply teamshifts/views.py:636]: `for app in apps: app.status = new_status` line already exists right after the `.update()` call and before the email-queueing loops, check lines after 658

File: teamshifts/static/teamshifts/applications.js (hunk 2 above)
Comment (MukundC25) [reply teamshifts/static/teamshifts/applications.js:44]: It's defined in `pretixcontrol/js/ui/confirmDialog.js`, loaded in `pretixcontrol/base.html` which teamshifts inherits from. Already used for the per-row status dropdown confirmation in one merged pr, here it just reuses it for the bulk buttons.

**Review by Aqil-Ahmad — APPROVED** [review]

**Review by prayag78 — APPROVED** [review]

**Review by SxxAq — APPROVED** [review]
LGTM

**Review by Saksham-Sirohi — APPROVED** [review]
